### ISSUE TITLE
incorrect usage of for else causes guaranteed duplication of addresses

### ISSUE THREAD

**Issue by aliomattux** [openlabs/magento_integration#241]
I see this module has not bee updated in over a year but for what its worth,

in partner.py there is a for each loop that checks customer addresses for a match.
see link https://github.com/openlabs/magento_integration/blob/develop/partner.py#L230

Python for else does not work the way described in the method description. No matter the result the else statement will always be called, meaning that in every case a new address will be created. Why call break? Why not just return the address that matches

**Comment by sharoonthomas** [comment]
@aliomattux yep, you are absolutely right. Do you want to send a pull request for the ?
